crates/core/tedge_api/src/mqtt_topics.rs

```diff
@@ -412,19 +540,20 @@ impl Display for Channel {
         match self {
             Channel::EntityMetadata => Ok(()),
 
-            Channel::Measurement { measurement_type } => write!(f, "m/{measurement_type}"),
+            Channel::Measurement { measurement_type } => write!(f, "/m/{measurement_type}"),
             Channel::MeasurementMetadata { measurement_type } => {
-                write!(f, "m/{measurement_type}/meta")
+                write!(f, "/m/{measurement_type}/meta")
             }
 
-            Channel::Event { event_type } => write!(f, "e/{event_type}"),
-            Channel::EventMetadata { event_type } => write!(f, "e/{event_type}/meta"),
+            Channel::Event { event_type } => write!(f, "/e/{event_type}"),
+            Channel::EventMetadata { event_type } => write!(f, "/e/{event_type}/meta"),
 
-            Channel::Alarm { alarm_type } => write!(f, "a/{alarm_type}"),
-            Channel::AlarmMetadata { alarm_type } => write!(f, "a/{alarm_type}/meta"),
+            Channel::Alarm { alarm_type } => write!(f, "/a/{alarm_type}"),
+            Channel::AlarmMetadata { alarm_type } => write!(f, "/a/{alarm_type}/meta"),
 
-            Channel::Command { operation, cmd_id } => write!(f, "cmd/{operation}/{cmd_id}"),
-            Channel::CommandMetadata { operation } => write!(f, "cmd/{operation}"),
+            Channel::Command { operation, cmd_id } => write!(f, "/cmd/{operation}/{cmd_id}"),
+            Channel::CommandMetadata { operation } => write!(f, "/cmd/{operation}"),
+            Channel::Health => write!(f, "/status/health"),
```

Comment:
Yes, here it was my bad because I wanted to remove the extra trailing slash in `Channel::EntityMetadata` case and didn't think about how `Display` impl makes it so `Channel` can be printed directly. Updated `topic_for` method so it handles the merging and added a test. Fixed in 2485a25.